pkg/crd/manifest.go

```diff
@@ -116,24 +156,22 @@ func (m *Manifest) ResourceAbsolutePath(path string) (string, error) {
 
 func checkCompleteness(m Manifest) error {
 	var omission string
-	if m.Istio == nil {
-		omission = "istio"
-	} else if m.Knative == nil {
-		omission = "knative"
-	} else if m.Namespace == nil {
-		omission = "namespace"
+	if m.Spec.Resources.System == nil {
```

Comment:
Validating the CR should be part of the CRD, instead of the runtime usage. For example https://github.com/knative/serving/blob/master/pkg/apis/serving/v1alpha1/service_validation.go

The amount of validation also needs to increase corresponding with the new capabilities in the CRD.